src/main/java/org/opensearch/searchrelevance/rest/RestPostExperimentAction.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.searchrelevance.rest;
+
+import static java.util.Collections.singletonList;
+import static org.opensearch.rest.RestRequest.Method.POST;
+import static org.opensearch.searchrelevance.common.PluginConstants.EXPERIMENTS_URI;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.searchrelevance.model.ExperimentType;
+import org.opensearch.searchrelevance.settings.SearchRelevanceSettingsAccessor;
+import org.opensearch.searchrelevance.transport.experiment.PostExperimentAction;
+import org.opensearch.searchrelevance.transport.experiment.PostExperimentRequest;
+import org.opensearch.searchrelevance.utils.ParserUtils;
+import org.opensearch.transport.client.node.NodeClient;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+/**
+ * Rest Action to facilitate requests to import a experiment.
+ */
+@AllArgsConstructor
+public class RestPostExperimentAction extends BaseRestHandler {
+    private static final String POST_EXPERIMENT_ACTION = "post_experiment_action";
+    private SearchRelevanceSettingsAccessor settingsAccessor;
+
+    @Override
+    public String getName() {
+        return POST_EXPERIMENT_ACTION;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return singletonList(new Route(POST, EXPERIMENTS_URI));
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        if (!settingsAccessor.isWorkbenchEnabled()) {
+            return channel -> channel.sendResponse(new BytesRestResponse(RestStatus.FORBIDDEN, "Search Relevance Workbench is disabled"));
+        }
+        XContentParser parser = request.contentParser();
+        Map<String, Object> source = parser.map();
+
+        String querySetId = (String) source.get("querySetId");
```

Comment:
please add validation for missing or blank values

Reply:
done...